File: inventeam_manufacturing_app/util/custom_picklist_script.py
import frappe

def validate_batch_qty(pick_list):
    for item in pick_list.locations:
        if item.batch_no:
            # Fetch the batch record
            batch = frappe.get_doc("Batch", item.batch_no)
            
            # Check if the batch quantity is less than the pick list quantity
            if batch.batch_qty > item.qty:
                frappe.throw(f"Batch quantity for {item.item_code} ({batch.batch_qty}) is less than the pick list quantity ({item.qty})")

def on_submit(doc, method):
    validate_batch_qty(doc)

File: inventeam_manufacturing_app/public/js/custom_pick_list_script.js
frappe.ui.form.on('Pick List', {
    validate: function(frm) {
        // Check if items exist in the Pick List
        if (frm.doc.locations && frm.doc.locations.length > 0) {
			console.log(frm.doc.locations.length);
            frm.doc.locations.forEach(function(item) {
                if (item.batch_no) {
                    // Item has a batch, check if the quantity matches the batch quantity
                    if (item.qty !== item.batch_qty) {
                        frappe.msgprint('Quantity does not match batch quantity for item ' + item.item_code);
                        frappe.validated = false;
                        return;
                    }
                }
            });
        }
    }
});